src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -8063,26 +8062,14 @@ def update_run_step(
             existing_step_run.update(step_run_update)
             session.add(existing_step_run)
 
-            # Update the output artifacts.
-            for name, artifact_version_id in step_run_update.outputs.items():
+            # Update the artifacts.
+            for name, artifact_version_id in list(
+                step_run_update.outputs.items()
+            ) + list(step_run_update.saved_artifact_versions.items()):
```

Comment:
Do we even need the `saved_artifact_versions` now? I guess this was just used to differentiate the output type for now, but this is not necessary anymore?